### PR TITLE
Pw orbital t implementation

### PR DESCRIPTION
**Pull request by walshmm**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

This adds PWOrbitalSetT as a templated version of PWOrbitalSet

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)
- Build related changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

built locally on a docker image of `ghcr.io/qmcpack/ubuntu22-openmpi:latest`

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes?. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- N/a. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- N/a. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.h [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+namespace qmcplusplus
+{
+
+template<class T>
+class PWOrbitalSetT : public SPOSetT<T>
+{
+  
+public:
+  using RealType = QMCTraits::RealType;
+  using ComplexType = QMCTraits::ComplexType;
```
Comment (quantumsteve) [inline src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.h]: Replace `ComplexType` with `T` and change `RealType` to
```suggestion
  using RealType = typename RealAlias_impl<T>::value_type;
```
( you need to `#include "type_traits/complex_help.hpp"`

File: src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    }
+  }
+}
+
+// Class concrete types from T
+// NOTE: This class only gets compiled if QMC_COMPLEX is defined, thus it is inherently complex
+// template class PWOrbitalSetT<double>;
+// template class PWOrbitalSetT<float>;
+template class PWOrbitalSetT<std::complex<double>>;
+// NOTE: MatrixOperators::product does not support floats
```
Comment (quantumsteve) [inline src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.cpp]: MatrixOperators::product requires A, B and C to be the same type, but that type can be double, std::complex<double>, float, or std::complex<float>.

File: src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.h (hunk 1 above)
Comment (quantumsteve) [reply src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.h]: @williamfgc Would you confirm if the above is correct?

Comment (walshmm) [reply src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.h]: I have altered the code to reflect this suggestion, still waiting for confirmation by @williamfgc

Comment (williamfgc) [reply src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.h]: Just reuse `SPOSet<T>::RealType`

**Comment by williamfgc** [comment]
@walshmm please squash the commits to one if possible

**Comment by walshmm** [comment]
@williamfgc there should be an option to "Squash and Merge" as a drop down option of the merge button.

**Comment by williamfgc** [comment]
@walshmm I'd rather you own that, so the commits comments reflect what your intent

**Pull request merged**